### PR TITLE
docs: add how to db and tx

### PR DESCRIPTION
**Pull request by Aflynn50**
Add docs on how to use the db and manage transactions.

### CODE REVIEW

File: docs/howto/query.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 (query)=
 # How to query the database
 
-## 1. Create/Make your types usable by SQLair
+## SQL-air proof your types
 The first step in building your query is to decide what you would like to put
```
Comment (tmihoc) [inline docs/howto/query.md:5]: SQLair-proof*

File: docs/howto/tx.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+(tx)=
+# How to manage SQLair transactions
+SQLair transactions are created on a `sqlair.DB`.
+
+> See more: {ref}`db`.
+## Begin a SQLair transaction
+
+To begin a transaction on your database call `DB.Begin`, passing a context and
```
Comment (tmihoc) [inline docs/howto/tx.md]:
```suggestion
To begin a transaction, on your database call `DB.Begin`, passing a context and
```

File: docs/howto/db.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+(db)=
+# How to use a SQLair database
```
Comment (tmihoc) [inline docs/howto/db.md:2]: In this entire db HTG the language is that of reference. Please convert to HTG language. (You do that in the tx HTG -- "To achieve x, do y. For example:" PS In some cases (e.g., the next paragraph) that will also significantly improve clarity.

File: docs/howto/query.md [hunk 4]
```diff
@@ -22,7 +22,7 @@ type Location struct {
 
 > See more: {ref}`types` 
 
-## 2. Write your query string using the types
+## Write your query string using the types
 
 SQLair queries are regular SQL queries with special input and output expressions
 which allow you to reference the Go types directly from the SQL.
```
Comment (tmihoc) [inline docs/howto/query.md]:
```suggestion
which allow you to reference the Go types directly from SQL.
```

File: docs/howto/query.md [hunk 5]
```diff
@@ -60,9 +65,11 @@ if err != nil {
...
-> See more [`sqlair.Prepare`](https://pkg.go.dev/github.com/canonical/sqlair#Prepare)
+> See more:
+[`sqlair.Prepare`](https://pkg.go.dev/github.com/canonical/sqlair#Prepare),
+[`sqlair.MustPrepare`](https://pkg.go.dev/github.com/canonical/sqlair#MustPrepare)
 
-## 4. Execute the statement on the database
+## Execute the statement on the database
 
 To execute the statement on a SQLair wrapped `DB` or a `TX`, use the `Query`
 method passing as parameters the `Statement` and all the input arguments
```
Comment (tmihoc) [inline docs/howto/query.md]:
```suggestion
method, passing as parameters the `Statement` and all the input arguments
```

File: docs/howto/query.md [hunk 6]
```diff
@@ -189,7 +196,7 @@ if err != nil {
 [`Query.Run`](https://pkg.go.dev/github.com/canonical/sqlair#Query.Run)
 
 
-## 5. (Optional) Get the query outcome
+## (Optional) Get the query outcome
 
 To get the outcome, provide a pointer to a `sqlair.Outcome` as the first
```
Comment (tmihoc) [inline docs/howto/query.md]:
```suggestion
To get the query outcome, use any of the `Get` methods, providing as a first argument a pointer to a `sqlair.Outcome` object.
```

File: docs/howto/tx.md (hunk 2 above)
Comment (tmihoc) [inline docs/howto/tx.md]:
```suggestion
To begin a transaction on your database, call `DB.Begin`, passing a context and
```

File: docs/howto/tx.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+if err != nil {
+    return err
+}
+```
+
+> See more: [DB.Begin](https://pkg.go.dev/github.com/canonical/sqlair#DB.Begin),
+[sqlair.TXOptions](https://pkg.go.dev/github.com/canonical/sqlair#TXOptions)
+
+## Query a SQLair transaction
+The transaction object has a `TX.Query` method that behaves the same as `DB.Query`. See
```
Comment (tmihoc) [inline docs/howto/tx.md]: Here you can cut out the text and just say "See ..."

File: docs/howto/tx.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+{ref}`query`.
+
+## Commit or rollback a SQLair transaction
+
+Use `TX.Commit` to commit a transaction or `TX.Rollback` to roll it back. Once a
```
Comment (tmihoc) [inline docs/howto/tx.md]:
```suggestion
To commit a transaction or to roll it back, use...
```

**Review by tmihoc — CHANGES_REQUESTED** [review]
Left some comments to make the style more how-to-guide-y.

File: docs/howto/query.md (hunk 4 above)
Comment (Aflynn50) [reply docs/howto/query.md]: Mean to say, I felt the numbers were a bit out of place in this article cuz we don't use them anywhere else in the how-tos plus making how to query a 5 step process sounds like quite a lot for a "convenience" library

File: docs/howto/db.md (hunk 3 above)
Comment (Aflynn50) [reply docs/howto/db.md:2]: Good point, changed the wording throughout

File: docs/howto/db.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+See {ref}`query` and {ref}`tx`.
+
+## Unwrap a SQLair database
+
+To unwrap a SQLair database and get out the `sql.DB`, use `DB.PlainDB`. SQLair
+does not handle closing the database, this should be done through the
+`database/sql` package.
+
+> See more:
+[`sqlair.PlainDB`](https://pkg.go.dev/github.com/canonical/sqlair#DB.PlainDB),
```
Comment (tmihoc) [inline docs/howto/db.md]:
```suggestion
[`sqlair.PlainDB`](https://pkg.go.dev/github.com/canonical/sqlair#PlainDB)
```

File: docs/howto/query.md [hunk 10]
```diff
@@ -60,12 +65,14 @@ if err != nil {
 }
 ```
 
-> See more [`sqlair.Prepare`](https://pkg.go.dev/github.com/canonical/sqlair#Prepare)
+> See more:
```
Comment (tmihoc) [inline docs/howto/query.md:68]: Looks like too much space. Please double-check.

File: docs/howto/tx.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+}
+```
+
+> See more: [DB.Begin](https://pkg.go.dev/github.com/canonical/sqlair#DB.Begin),
+[sqlair.TXOptions](https://pkg.go.dev/github.com/canonical/sqlair#TXOptions)
+
+## Query a SQLair transaction
+See {ref}`query`.
+
+## Commit or rollback a SQLair transaction
```
Comment (tmihoc) [inline docs/howto/tx.md]: rollback or roll back?

File: docs/howto/tx.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+To commit a transaction or roll it back, use `TX.Commit` or `TX.Rollback`. Once
+a transaction has been committed or rolled back any further operations on it
```
Comment (tmihoc) [inline docs/howto/tx.md]:
```suggestion
a transaction has been committed or rolled back, any further operations on it
```

File: docs/howto/query.md [hunk 13]
```diff
@@ -1,7 +1,7 @@
 (query)=
 # How to query the database
 
-## 1. Create/Make your types usable by SQLair
+## SQLair-proof your types
```
Comment (tmihoc) [inline docs/howto/query.md:4]:
```suggestion
## 1. SQLair-proof your types
```

File: docs/howto/query.md [hunk 14]
```diff
@@ -22,10 +22,10 @@ type Location struct {
 
 > See more: {ref}`types` 
 
-## 2. Write your query string using the types
+## Write your query string using the types
```
Comment (tmihoc) [inline docs/howto/query.md:25]:
```suggestion
## 2.Write your query string using the types
```

File: docs/howto/query.md [hunk 15]
```diff
@@ -41,7 +41,7 @@ query := `
 
 > See more {ref}`input-expression-syntax`, {ref}`output-expression-syntax`.
 
-## 3. Prepare your query string to get a Statement
+## Prepare your query string to get a statement
```
Comment (tmihoc) [inline docs/howto/query.md:44]:
```suggestion
## 3. Prepare your query string to get a statement
```

File: docs/howto/query.md [hunk 16]
```diff
@@ -60,12 +65,14 @@ if err != nil {
 }
 ```
 
-> See more [`sqlair.Prepare`](https://pkg.go.dev/github.com/canonical/sqlair#Prepare)
+> See more:
+[`sqlair.Prepare`](https://pkg.go.dev/github.com/canonical/sqlair#Prepare),
+[`sqlair.MustPrepare`](https://pkg.go.dev/github.com/canonical/sqlair#MustPrepare)
 
-## 4. Execute the statement on the database
+## Execute the statement on the database
```
Comment (tmihoc) [inline docs/howto/query.md:72]:
```suggestion
## 4. Execute the statement on the database
```

File: docs/howto/query.md [hunk 17]
```diff
@@ -189,11 +196,11 @@ if err != nil {
 [`Query.Run`](https://pkg.go.dev/github.com/canonical/sqlair#Query.Run)
 
 
-## 5. (Optional) Get the query outcome
+## (Optional) Get the query outcome
```
Comment (tmihoc) [inline docs/howto/query.md:199]:
```suggestion
## 5. (Optional) Get the query outcome
```

File: docs/howto/db.md [hunk 18]
```diff
@@ -0,0 +1,29 @@
+(db)=
+# How to use a SQLair database
+To create a SQLair database and run SQLair queries on it, you need to wrap a
+`sql.DB` database object from the `database/sql` package. SQLair is a
+convenience wrapper around the database, it does not create the DB itself.
+Operations such as opening, closing and managing the DB settings is done through
```
Comment (tmihoc) [inline docs/howto/db.md]:
```suggestion
Operations such as opening, closing and managing the DB settings are done through
```

File: docs/howto/db.md [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+[`database/sql.DB`](https://pkg.go.dev/database/sql#DB)
+
+## Query a SQLair database
+
+See {ref}`query` and {ref}`tx`.
+
+## Unwrap a SQLair database
+
+To unwrap a SQLair database and get out the `sql.DB`, use `DB.PlainDB`. SQLair
+does not handle closing the database, this should be done through the
```
Comment (tmihoc) [inline docs/howto/db.md]:
```suggestion
does not handle closing the database -- this should be done through the
```

File: docs/howto/query.md [hunk 20]
```diff
@@ -22,10 +22,10 @@ type Location struct {
 
 > See more: {ref}`types` 
 
-## 2. Write your query string using the types
+## Write your query string using the types
 
 SQLair queries are regular SQL queries with special input and output expressions
-which allow you to reference the Go types directly from the SQL.
+which allow you to reference the Go types directly from SQL.
```
Comment (tmihoc) [inline docs/howto/query.md]:
```suggestion
that allow you to reference the Go types directly from SQL.
```

**Review by tmihoc — APPROVED** [review]
Thanks for making the changes! Made some more suggestions. In addition to those:

> To iterate over the rows returned from the query, get an `Iterator` with use
`Query.Iter`.

There's something wrong with this sentence, at the end.

File: docs/howto/db.md (hunk 9 above)
Comment (Aflynn50) [reply docs/howto/db.md]: Ah, this is the other side that should change, sqlair.PlainDB -> DB.PlainDB

File: docs/howto/query.md (hunk 10 above)
Comment (Aflynn50) [reply docs/howto/query.md:68]: If they're in the same paragraph they are rendered as one block, I think this should be fine

File: docs/howto/tx.md (hunk 11 above)
Comment (Aflynn50) [reply docs/howto/tx.md]: I'll go with roll back

File: docs/howto/query.md (hunk 20 above)
Comment (Aflynn50) [reply docs/howto/query.md]: Also changed allow to enable.

**Pull request merged**